// @ruvice/my-maple-models/dist/utils/constants.js
import { ItemEquipOption, SymbolEquipOption } from "../models/maple/Equipment/equipmentTypes";
import { CharacterClass, StatName } from "../models/maple/Generic/enums";
export const CharacterClassLocalisationSEA = {
    [CharacterClass.Hero]: "Hero",
    [CharacterClass.Paladin]: "Paladin",
    [CharacterClass.DarkKnight]: "Dark Knight",
    [CharacterClass.SoulMaster]: "Soul Master",
    [CharacterClass.Mihile]: "Mihile",
    [CharacterClass.Blaster]: "Blaster",
    [CharacterClass.DemonSlayer]: "Demon Slayer",
    [CharacterClass.DemonAvenger]: "Demon Avenger",
    [CharacterClass.Aran]: "Aran",
    [CharacterClass.Kaiser]: "Kaiser",
    [CharacterClass.Adele]: "Adele",
    [CharacterClass.Zero]: "Zero",
    [CharacterClass.ArchMageFP]: "ArchMage F/P",
    [CharacterClass.ArchMageIL]: "ArchMage I/L",
    [CharacterClass.Bishop]: "Bishop",
    [CharacterClass.FlameWizard]: "Flame Wizard",
    [CharacterClass.BattleMage]: "Battle Mage",
    [CharacterClass.Evan]: "Evan",
    [CharacterClass.Luminous]: "Luminous",
    [CharacterClass.Illium]: "Illium",
    [CharacterClass.Lara]: "Lara",
    [CharacterClass.Kinesis]: "Kinesis",
    [CharacterClass.Bowmaster]: "Bowmaster",
    [CharacterClass.CrossbowMaster]: "Crossbow Master",
    [CharacterClass.Pathfinder]: "Pathfinder",
    [CharacterClass.WindBreaker]: "Wind Breaker",
    [CharacterClass.WildHunter]: "Wild Hunter",
    [CharacterClass.Mercedes]: "Mercedes",
    [CharacterClass.Kaine]: "Kaine",
    [CharacterClass.NightLord]: "Night Lord",
    [CharacterClass.Shadower]: "Shadower",
    [CharacterClass.BladeMaster]: "Blade Master",
    [CharacterClass.NightWalker]: "Night Walker",
    [CharacterClass.Xenon]: "Xenon",
    [CharacterClass.Phantom]: "Phantom",
    [CharacterClass.Cadena]: "Cadena",
    [CharacterClass.Khali]: "Khali",
    [CharacterClass.HoYoung]: "HoYoung",
    [CharacterClass.Viper]: "Viper",
    [CharacterClass.Captain]: "Captain",
    [CharacterClass.CannonMaster]: "Cannon Master",
    [CharacterClass.Striker]: "Striker",
    [CharacterClass.Mechanic]: "Mechanic",
    [CharacterClass.Eunwol]: "Eunwol",
    [CharacterClass.AngelicBuster]: "Angelic Buster",
    [CharacterClass.Ark]: "Ark",
    [CharacterClass.Hayato]: "Hayato",
    [CharacterClass.Kanna]: "Kanna",
    [CharacterClass.Len]: "Len"
};
export const CharacterClassLocalisationKMS = {
    [CharacterClass.Hero]: "히어로",
    [CharacterClass.DarkKnight]: "다크나이트",
    [CharacterClass.Paladin]: "팔라딘",
    [CharacterClass.Bishop]: "비숍",
    [CharacterClass.ArchMageFP]: "불독",
    [CharacterClass.ArchMageIL]: "썬콜",
    [CharacterClass.Shadower]: "섀도어",
    [CharacterClass.NightLord]: "나이트로드",
    [CharacterClass.BladeMaster]: "듀얼블레이드",
    [CharacterClass.Bowmaster]: "보우마스터",
    [CharacterClass.Pathfinder]: "패스파인더",
    [CharacterClass.SoulMaster]: "소울마스터",
    [CharacterClass.FlameWizard]: "플레임위자드",
    [CharacterClass.NightWalker]: "나이트워커",
    [CharacterClass.WindBreaker]: "윈드브레이커",
    [CharacterClass.Striker]: "스트라이커",
    [CharacterClass.Mihile]: "미하일",
    [CharacterClass.Aran]: "아란",
    [CharacterClass.Evan]: "에반",
    [CharacterClass.Mercedes]: "메르세데스",
    [CharacterClass.Luminous]: "루미너스",
    [CharacterClass.Phantom]: "팬텀",
    [CharacterClass.Eunwol]: "은월",
    [CharacterClass.AngelicBuster]: "엔젤릭버스터",
    [CharacterClass.Kaiser]: "카이저",
    [CharacterClass.Cadena]: "카데나",
    [CharacterClass.Kaine]: "카인",
    [CharacterClass.Xenon]: "제논",
    [CharacterClass.Mechanic]: "메카닉",
    [CharacterClass.BattleMage]: "배틀메이지",
    [CharacterClass.WildHunter]: "와일드헌터",
    [CharacterClass.Blaster]: "블래스터",
    [CharacterClass.DemonAvenger]: "데몬 어벤저",
    [CharacterClass.DemonSlayer]: "데몬 슬레이어",
    [CharacterClass.Adele]: "아델",
    [CharacterClass.Ark]: "아크",
    [CharacterClass.Illium]: "일리움",
    [CharacterClass.Khali]: "칼리",
    [CharacterClass.HoYoung]: "호영",
    [CharacterClass.Lara]: "라라",
    [CharacterClass.Kinesis]: "키네시스",
    [CharacterClass.Zero]: "제로",
    [CharacterClass.CrossbowMaster]: "신궁",
    [CharacterClass.Viper]: "바이퍼",
    [CharacterClass.Captain]: "캡틴",
    [CharacterClass.CannonMaster]: "캐논슈터",
    [CharacterClass.Hayato]: "",
    [CharacterClass.Kanna]: "",
    [CharacterClass.Len]: "렌"
};
export const ITEM_EQUIP_STAT_OPTIONS_SEA = {
    [ItemEquipOption.Str]: 'STR :',
    [ItemEquipOption.Dex]: 'DEX :',
    [ItemEquipOption.Int]: 'INT :',
    [ItemEquipOption.Luk]: 'LUK :',
    [ItemEquipOption.MaxHp]: 'MaxHP :',
    [ItemEquipOption.MaxMp]: 'MaxMP :',
    [ItemEquipOption.MaxHpRate]: 'HP :',
    [ItemEquipOption.MaxMpRate]: 'MP :',
    [ItemEquipOption.AttackPower]: 'WEAPON ATTACK :',
    [ItemEquipOption.MagicPower]: 'MAGIC ATTACK :',
    [ItemEquipOption.Armor]: 'DEF :',
    [ItemEquipOption.BossDamage]: 'When attacking bosses, damage ',
    [ItemEquipOption.IgnoreMonsterArmor]: 'Ignore Monster Defense :',
    [ItemEquipOption.Damage]: 'Total Damage:',
    [ItemEquipOption.Speed]: 'SPEED :',
    [ItemEquipOption.Jump]: 'JUMP :',
    [ItemEquipOption.AllStat]: 'All Stats :',
    [ItemEquipOption.EquipmentLevelDecrease]: 'Wearer \'s Level is reduced : '
};
export const ITEM_EQUIP_STAT_OPTIONS_KMS = {
    [ItemEquipOption.Str]: 'STR :',
    [ItemEquipOption.Dex]: 'DEX :',
    [ItemEquipOption.Int]: 'INT :',
    [ItemEquipOption.Luk]: 'LUK :',
    [ItemEquipOption.MaxHp]: '최대 HP :',
    [ItemEquipOption.MaxMp]: '최대 MP :',
    [ItemEquipOption.MaxHpRate]: '최대 HP(%) :',
    [ItemEquipOption.MaxMpRate]: '최대 MP(%) :',
    [ItemEquipOption.AttackPower]: '공격력 :',
    [ItemEquipOption.MagicPower]: '마력 :',
    [ItemEquipOption.Armor]: '방어력 :',
    [ItemEquipOption.BossDamage]: '보스 공격 시 데미지 증가(%) ',
    [ItemEquipOption.IgnoreMonsterArmor]: '몬스터 방어율 무시(%) : ',
    [ItemEquipOption.Damage]: '데미지(%):',
    [ItemEquipOption.Speed]: '이동속도:',
    [ItemEquipOption.Jump]: '점프력 :',
    [ItemEquipOption.AllStat]: '올스탯(%) :',
    [ItemEquipOption.EquipmentLevelDecrease]: '착용 레벨 감소 : ',
};
export const SYMBOL_OPTIONS_SEA = {
    [SymbolEquipOption.SymbolLevel]: 'Growth Level  :',
    [SymbolEquipOption.Growth]: 'Growth :',
    [SymbolEquipOption.Str]: 'STR : +',
    [SymbolEquipOption.Dex]: 'DEX : +',
    [SymbolEquipOption.Int]: 'INT : +',
    [SymbolEquipOption.Luk]: 'LUK : +',
    [SymbolEquipOption.Hp]: 'HP : +',
    [SymbolEquipOption.ExpRate]: 'EXP OBTAINED : +',
    [SymbolEquipOption.MesoRate]: 'MESO DROP RATE : +',
    [SymbolEquipOption.DropRate]: 'Item Drop Rate : +'
};
export const SYMBOL_OPTIONS_KMS = {
    [SymbolEquipOption.SymbolLevel]: '심볼 레벨 :',
    [SymbolEquipOption.Growth]: '성장치 :',
    [SymbolEquipOption.Str]: 'STR : +',
    [SymbolEquipOption.Dex]: 'DEX : +',
    [SymbolEquipOption.Int]: 'INT : +',
    [SymbolEquipOption.Luk]: 'LUK : +',
    [SymbolEquipOption.Hp]: 'HP : +',
    [SymbolEquipOption.ExpRate]: '경험치 획득량 : +',
    [SymbolEquipOption.MesoRate]: '메소 획득량 : +',
    [SymbolEquipOption.DropRate]: '아이템 드롭률 : +'
};
export const STAT_SEA = {
    [StatName.CombatPower]: "Combat Power",
    [StatName.Hp]: "HP",
    [StatName.Mp]: "MP",
    [StatName.Str]: "STR",
    [StatName.Dex]: "DEX",
    [StatName.Int]: "INT",
    [StatName.Luk]: "LUK",
    [StatName.StatAtt]: "Stat Att",
    [StatName.Damage]: "Damage",
    [StatName.FinalDamage]: "Final Damage",
    [StatName.BossMonsterDamage]: "Boss Monster Damage",
    [StatName.IgnoreDef]: "Ignore DEF",
    [StatName.NormalMonsterDamage]: "Normal Monster Damage",
    [StatName.Att]: "ATT",
    [StatName.CriticalRate]: "Critical Rate",
    [StatName.MAtt]: "MATT",
    [StatName.CriticalDamage]: "Critical Damage",
    [StatName.CooldownReduction]: "Cooldown Reduction",
    [StatName.BuffDuration]: "Buff Duration",
    [StatName.UnaffectedByCooldown]: "Unaffected by Cooldown",
    [StatName.IgnoreElementalResistance]: "Ignore Elemental Resistance",
    [StatName.AbnormalStatusAdditionalDamage]: "Abnormal Status Add. Damage",
    [StatName.SummonDurationIncrease]: "Summon Duration Increase",
    [StatName.MesoObtained]: "Meso Obtained",
    [StatName.Starforce]: "Star Force",
    [StatName.ItemDropRate]: "Item Drop Rate",
    [StatName.ArcaneForce]: "Arcane Force",
    [StatName.AdditionalExpObtained]: "Additional EXP Obtained",
    [StatName.AuthenticForce]: "Authentic Force",
    [StatName.Def]: "DEF",
    [StatName.StatusResistance]: "Status Resistance",
    [StatName.Speed]: "Speed",
    [StatName.Jump]: "Jump",
    [StatName.Stance]: "Stance",
    [StatName.AttackSpeed]: "Attack Speed",
    [StatName.MinimumAttStat]: "Minimum ATT",
    [StatName.MaximumAttStat]: "Maximum ATT",
    [StatName.CooldownReductionSec]: "Cooldown Reduction (sec)",
    [StatName.CooldownReductionPercent]: "Cooldown Reduction (%)"
};
export const STAT_KMS = {
    [StatName.CombatPower]: "전투력",
    [StatName.Hp]: "HP",
    [StatName.Mp]: "MP",
    [StatName.Str]: "STR",
    [StatName.Dex]: "DEX",
    [StatName.Int]: "INT",
    [StatName.Luk]: "LUK",
    [StatName.StatAtt]: "스탯공격력", // Not in translations; using partial match or fallback
    [StatName.Damage]: "데미지",
    [StatName.FinalDamage]: "최종 데미지",
    [StatName.BossMonsterDamage]: "보스 몬스터 데미지",
    [StatName.IgnoreDef]: "방어율 무시",
    [StatName.NormalMonsterDamage]: "일반 몬스터 데미지",
    [StatName.Att]: "공격력",
    [StatName.CriticalRate]: "크리티컬 확률",
    [StatName.MAtt]: "마력",
    [StatName.CriticalDamage]: "크리티컬 데미지",
    [StatName.CooldownReduction]: "재사용 대기시간 감소 (%)",
    [StatName.BuffDuration]: "버프 지속시간",
    [StatName.UnaffectedByCooldown]: "재사용 대기시간 미적용",
    [StatName.IgnoreElementalResistance]: "속성 내성 무시",
    [StatName.AbnormalStatusAdditionalDamage]: "상태이상 추가 데미지",
    [StatName.SummonDurationIncrease]: "소환수 지속시간 증가",
    [StatName.MesoObtained]: "메소 획득량",
    [StatName.Starforce]: "스타포스",
    [StatName.ItemDropRate]: "아이템 드롭률",
    [StatName.ArcaneForce]: "아케인포스",
    [StatName.AdditionalExpObtained]: "추가 경험치 획득",
    [StatName.AuthenticForce]: "어센틱포스",
    [StatName.Def]: "방어력",
    [StatName.StatusResistance]: "상태이상 내성",
    [StatName.Speed]: "이동속도",
    [StatName.Jump]: "점프력",
    [StatName.Stance]: "스탠스",
    [StatName.AttackSpeed]: "공격 속도",
    [StatName.MinimumAttStat]: "최소 스탯공격력",
    [StatName.MaximumAttStat]: "최대 스탯공격력",
    [StatName.CooldownReductionSec]: "재사용 대기시간 감소 (초)",
    [StatName.CooldownReductionPercent]: "재사용 대기시간 감소 (%)"
};
