// @ruvice/my-maple-models/dist/models/maple/Generic/enums.js
export var Potential;
(function (Potential) {
    Potential["None"] = "None";
    Potential["Rare"] = "Rare";
    Potential["Epic"] = "Epic";
    Potential["Unique"] = "Unique";
    Potential["Legendary"] = "Legendary";
})(Potential || (Potential = {}));
export var CharacterClass;
(function (CharacterClass) {
    CharacterClass["Hero"] = "Hero";
    CharacterClass["Paladin"] = "Paladin";
    CharacterClass["DarkKnight"] = "DarkKnight";
    CharacterClass["SoulMaster"] = "SoulMaster";
    CharacterClass["Mihile"] = "Mihile";
    CharacterClass["Blaster"] = "Blaster";
    CharacterClass["DemonSlayer"] = "DemonSlayer";
    CharacterClass["DemonAvenger"] = "DemonAvenger";
    CharacterClass["Aran"] = "Aran";
    CharacterClass["Kaiser"] = "Kaiser";
    CharacterClass["Adele"] = "Adele";
    CharacterClass["Zero"] = "Zero";
    CharacterClass["ArchMageFP"] = "ArchMageFP";
    CharacterClass["ArchMageIL"] = "ArchMageIL";
    CharacterClass["Bishop"] = "Bishop";
    CharacterClass["FlameWizard"] = "FlameWizard";
    CharacterClass["BattleMage"] = "BattleMage";
    CharacterClass["Evan"] = "Evan";
    CharacterClass["Luminous"] = "Luminous";
    CharacterClass["Illium"] = "Illium";
    CharacterClass["Lara"] = "Lara";
    CharacterClass["Kinesis"] = "Kinesis";
    CharacterClass["Bowmaster"] = "Bowmaster";
    CharacterClass["CrossbowMaster"] = "CrossbowMaster";
    CharacterClass["Pathfinder"] = "Pathfinder";
    CharacterClass["WindBreaker"] = "WindBreaker";
    CharacterClass["WildHunter"] = "WildHunter";
    CharacterClass["Mercedes"] = "Mercedes";
    CharacterClass["Kaine"] = "Kaine";
    CharacterClass["NightLord"] = "NightLord";
    CharacterClass["Shadower"] = "Shadower";
    CharacterClass["BladeMaster"] = "BladeMaster";
    CharacterClass["NightWalker"] = "NightWalker";
    CharacterClass["Xenon"] = "Xenon";
    CharacterClass["Phantom"] = "Phantom";
    CharacterClass["Cadena"] = "Cadena";
    CharacterClass["Khali"] = "Khali";
    CharacterClass["HoYoung"] = "HoYoung";
    CharacterClass["Viper"] = "Viper";
    CharacterClass["Captain"] = "Captain";
    CharacterClass["CannonMaster"] = "CannonMaster";
    CharacterClass["Striker"] = "Striker";
    CharacterClass["Mechanic"] = "Mechanic";
    CharacterClass["Eunwol"] = "Eunwol";
    CharacterClass["AngelicBuster"] = "AngelicBuster";
    CharacterClass["Ark"] = "Ark";
    CharacterClass["Hayato"] = "Hayato";
    CharacterClass["Kanna"] = "Kanna";
})(CharacterClass || (CharacterClass = {}));
