// @ruvice/my-maple-models/dist/models/api/apiResponseParser.js
import { ItemEquipmentSlot } from "../maple/Equipment/equipmentTypes";
import { Potential, CharacterClass, StatName, MapleServer } from "../maple/Generic/enums";
export const ItemEquipmentSlotMappingSEA = {
    "Ring1": ItemEquipmentSlot.Ring1,
    "Ring2": ItemEquipmentSlot.Ring2,
    "Ring3": ItemEquipmentSlot.Ring3,
    "Ring4": ItemEquipmentSlot.Ring4,
    "Pendant": ItemEquipmentSlot.Pendant,
    "Pendant 2": ItemEquipmentSlot.Pendant2,
    "Weapon": ItemEquipmentSlot.Weapon,
    "Belt": ItemEquipmentSlot.Belt,
    "Hat": ItemEquipmentSlot.Hat,
    "Face Acc.": ItemEquipmentSlot.FaceAccessory,
    "Eye Acc.": ItemEquipmentSlot.EyeAccessory,
    "Top": ItemEquipmentSlot.Top,
    "Bottom": ItemEquipmentSlot.Bottom,
    "Shoes": ItemEquipmentSlot.Shoes,
    "Earring": ItemEquipmentSlot.Earring,
    "Shoulder": ItemEquipmentSlot.ShoulderDecoration,
    "Glove": ItemEquipmentSlot.Glove,
    "Emblem": ItemEquipmentSlot.Emblem,
    "Badge": ItemEquipmentSlot.Badge,
    "Medal": ItemEquipmentSlot.Medal,
    "Secondary Weapons": ItemEquipmentSlot.SecondaryWeapons,
    "Cape": ItemEquipmentSlot.Cape,
    "M. Heart": ItemEquipmentSlot.MechanicalHeart,
    "Pocket Item": ItemEquipmentSlot.PocketItem
};
export const ItemEquipmentSlotMappingKMS = {
    "모자": ItemEquipmentSlot.Hat,
    "상의": ItemEquipmentSlot.Top,
    "하의": ItemEquipmentSlot.Bottom,
    "신발": ItemEquipmentSlot.Shoes,
    "장갑": ItemEquipmentSlot.Glove,
    "망토": ItemEquipmentSlot.Cape,
    "무기": ItemEquipmentSlot.Weapon,
    "보조무기": ItemEquipmentSlot.SecondaryWeapons,
    "엠블렘": ItemEquipmentSlot.Emblem,
    "귀고리": ItemEquipmentSlot.Earring,
    "벨트": ItemEquipmentSlot.Belt,
    "펜던트": ItemEquipmentSlot.Pendant,
    "펜던트2": ItemEquipmentSlot.Pendant2,
    "반지1": ItemEquipmentSlot.Ring1,
    "반지2": ItemEquipmentSlot.Ring2,
    "반지3": ItemEquipmentSlot.Ring3,
    "반지4": ItemEquipmentSlot.Ring4,
    "어깨장식": ItemEquipmentSlot.ShoulderDecoration,
    "훈장": ItemEquipmentSlot.Medal,
    "뱃지": ItemEquipmentSlot.Badge,
    "포켓 아이템": ItemEquipmentSlot.PocketItem,
    "기계 심장": ItemEquipmentSlot.MechanicalHeart,
    "얼굴장식": ItemEquipmentSlot.FaceAccessory,
    "눈장식": ItemEquipmentSlot.EyeAccessory
};
export const PotentialMappingSEA = {
    "Rare": Potential.Rare,
    "Epic": Potential.Epic,
    "Unique": Potential.Unique,
    "Legendary": Potential.Legendary
};
export const PotentialMappingKMS = {
    "레어": Potential.Rare,
    "에픽": Potential.Epic,
    "유니크": Potential.Unique,
    "레전드리": Potential.Legendary
};
export const CharacterClassMappingSEA = {
    "Hero": CharacterClass.Hero,
    "Paladin": CharacterClass.Paladin,
    "Dark Knight": CharacterClass.DarkKnight,
    "Soul Master": CharacterClass.SoulMaster,
    "Mihile": CharacterClass.Mihile,
    "Blaster": CharacterClass.Blaster,
    "Demon Slayer": CharacterClass.DemonSlayer,
    "Demon Avenger": CharacterClass.DemonAvenger,
    "Aran": CharacterClass.Aran,
    "Kaiser": CharacterClass.Kaiser,
    "Adele": CharacterClass.Adele,
    "Zero": CharacterClass.Zero,
    "ArchMage F/P": CharacterClass.ArchMageFP,
    "ArchMage I/L": CharacterClass.ArchMageIL,
    "Bishop": CharacterClass.Bishop,
    "Flame Wizard": CharacterClass.FlameWizard,
    "Battle Mage": CharacterClass.BattleMage,
    "Evan": CharacterClass.Evan,
    "Luminous": CharacterClass.Luminous,
    "Illium": CharacterClass.Illium,
    "Lara": CharacterClass.Lara,
    "Kinesis": CharacterClass.Kinesis,
    "Bowmaster": CharacterClass.Bowmaster,
    "Crossbow Master": CharacterClass.CrossbowMaster,
    "Pathfinder": CharacterClass.Pathfinder,
    "Wind Breaker": CharacterClass.WindBreaker,
    "Wild Hunter": CharacterClass.WildHunter,
    "Mercedes": CharacterClass.Mercedes,
    "Kaine": CharacterClass.Kaine,
    "Night Lord": CharacterClass.NightLord,
    "Shadower": CharacterClass.Shadower,
    "Blade Master": CharacterClass.BladeMaster,
    "Night Walker": CharacterClass.NightWalker,
    "Xenon": CharacterClass.Xenon,
    "Phantom": CharacterClass.Phantom,
    "Cadena": CharacterClass.Cadena,
    "Khali": CharacterClass.Khali,
    "HoYoung": CharacterClass.HoYoung,
    "Viper": CharacterClass.Viper,
    "Captain": CharacterClass.Captain,
    "Cannon Master": CharacterClass.CannonMaster,
    "Striker": CharacterClass.Striker,
    "Mechanic": CharacterClass.Mechanic,
    "Eunwol": CharacterClass.Eunwol,
    "Angelic Buster": CharacterClass.AngelicBuster,
    "Ark": CharacterClass.Ark,
    "Hayato": CharacterClass.Hayato,
    "Kanna": CharacterClass.Kanna,
    "Len": CharacterClass.Len
};
export const CharacterClassMappingKMS = {
    "히어로": CharacterClass.Hero,
    "다크나이트": CharacterClass.DarkKnight,
    "팔라딘": CharacterClass.Paladin,
    "비숍": CharacterClass.Bishop,
    "불독": CharacterClass.ArchMageFP,
    "썬콜": CharacterClass.ArchMageIL,
    "섀도어": CharacterClass.Shadower,
    "나이트로드": CharacterClass.NightLord,
    "듀얼블레이드": CharacterClass.BladeMaster,
    "보우마스터": CharacterClass.Bowmaster,
    "패스파인더": CharacterClass.Pathfinder,
    "소울마스터": CharacterClass.SoulMaster,
    "플레임위자드": CharacterClass.FlameWizard,
    "나이트워커": CharacterClass.NightWalker,
    "윈드브레이커": CharacterClass.WindBreaker,
    "스트라이커": CharacterClass.Striker,
    "미하일": CharacterClass.Mihile,
    "아란": CharacterClass.Aran,
    "에반": CharacterClass.Evan,
    "메르세데스": CharacterClass.Mercedes,
    "루미너스": CharacterClass.Luminous,
    "팬텀": CharacterClass.Phantom,
    "은월": CharacterClass.Eunwol,
    "엔젤릭버스터": CharacterClass.AngelicBuster,
    "카이저": CharacterClass.Kaiser,
    "카데나": CharacterClass.Cadena,
    "카인": CharacterClass.Kaine,
    "제논": CharacterClass.Xenon,
    "메카닉": CharacterClass.Mechanic,
    "배틀메이지": CharacterClass.BattleMage,
    "와일드헌터": CharacterClass.WildHunter,
    "블래스터": CharacterClass.Blaster,
    "데몬 어벤저": CharacterClass.DemonAvenger,
    "데몬 슬레이어": CharacterClass.DemonSlayer,
    "아델": CharacterClass.Adele,
    "아크": CharacterClass.Ark,
    "일리움": CharacterClass.Illium,
    "칼리": CharacterClass.Khali,
    "호영": CharacterClass.HoYoung,
    "라라": CharacterClass.Lara,
    "키네시스": CharacterClass.Kinesis,
    "제로": CharacterClass.Zero,
    "신궁": CharacterClass.CrossbowMaster,
    "바이퍼": CharacterClass.Viper,
    "캡틴": CharacterClass.Captain,
    "캐논슈터": CharacterClass.CannonMaster,
    "렌": CharacterClass.Len
};
export const StatMappingSEA = {
    "Minimum ATT Stat": StatName.MinimumAttStat,
    "Maximum ATT Stat": StatName.MaximumAttStat,
    "Damage": StatName.Damage,
    "Boss Monster Damage": StatName.BossMonsterDamage,
    "Final Damage": StatName.FinalDamage,
    "Ignore DEF": StatName.IgnoreDef,
    "Critical Rate": StatName.CriticalRate,
    "Critical Damage": StatName.CriticalDamage,
    "Status Resistance": StatName.StatusResistance,
    "Stance": StatName.Stance,
    "DEF": StatName.Def,
    "Speed": StatName.Speed,
    "Jump": StatName.Jump,
    "Star Force": StatName.Starforce,
    "Arcane Force": StatName.ArcaneForce,
    "Authentic Force": StatName.AuthenticForce,
    "STR": StatName.Str,
    "DEX": StatName.Dex,
    "INT": StatName.Int,
    "LUK": StatName.Luk,
    "HP": StatName.Hp,
    "MP": StatName.Mp,
    "Item Drop Rate": StatName.ItemDropRate,
    "Meso Obtained": StatName.MesoObtained,
    "Buff Duration": StatName.BuffDuration,
    "Attack Speed": StatName.AttackSpeed,
    "Normal monster damage": StatName.NormalMonsterDamage,
    "Cooldown reduction (sec)": StatName.CooldownReductionSec,
    "Cooldown reduction (%)": StatName.CooldownReductionPercent,
    "Unaffected by Cooldown": StatName.UnaffectedByCooldown,
    "Ignore Elemental Resistance": StatName.IgnoreElementalResistance,
    "Abnormal Status Additional Damage": StatName.AbnormalStatusAdditionalDamage,
    "Additonal EXP Obtained": StatName.AdditionalExpObtained,
    "ATT": StatName.Att,
    "MATT": StatName.MAtt,
    "Combat Power": StatName.CombatPower,
    "Summon Duration Increase": StatName.SummonDurationIncrease
};
export const StatMappingKMS = {
    "최소 스탯공격력": StatName.MinimumAttStat,
    "최대 스탯공격력": StatName.MaximumAttStat,
    "데미지": StatName.Damage,
    "보스 몬스터 데미지": StatName.BossMonsterDamage,
    "최종 데미지": StatName.FinalDamage,
    "방어율 무시": StatName.IgnoreDef,
    "크리티컬 확률": StatName.CriticalRate,
    "크리티컬 데미지": StatName.CriticalDamage,
    "상태이상 내성": StatName.StatusResistance,
    "스탠스": StatName.Stance,
    "방어력": StatName.Def,
    "이동속도": StatName.Speed,
    "점프력": StatName.Jump,
    "스타포스": StatName.Starforce,
    "아케인포스": StatName.ArcaneForce,
    "어센틱포스": StatName.AuthenticForce,
    "STR": StatName.Str,
    "DEX": StatName.Dex,
    "INT": StatName.Int,
    "LUK": StatName.Luk,
    "HP": StatName.Hp,
    "MP": StatName.Mp,
    "아이템 드롭률": StatName.ItemDropRate,
    "메소 획득량": StatName.MesoObtained,
    "버프 지속시간": StatName.BuffDuration,
    "공격 속도": StatName.AttackSpeed,
    "일반 몬스터 데미지": StatName.NormalMonsterDamage,
    "재사용 대기시간 감소 (초)": StatName.CooldownReductionSec,
    "재사용 대기시간 감소 (%)": StatName.CooldownReductionPercent,
    "재사용 대기시간 미적용": StatName.UnaffectedByCooldown,
    "속성 내성 무시": StatName.IgnoreElementalResistance,
    "상태이상 추가 데미지": StatName.AbnormalStatusAdditionalDamage,
    "추가 경험치 획득": StatName.AdditionalExpObtained,
    "공격력": StatName.Att,
    "마력": StatName.MAtt,
    "전투력": StatName.CombatPower,
    "소환수 지속시간 증가": StatName.SummonDurationIncrease
};
export function toMapleServer(input) {
    return Object.values(MapleServer).includes(input)
        ? input
        : undefined;
}
