// @ruvice/my-maple-models/dist/models/maple/Equipment/equipmentTypes.js
export var ItemEquipmentSlot;
(function (ItemEquipmentSlot) {
    ItemEquipmentSlot["Ring1"] = "Ring1";
    ItemEquipmentSlot["Ring2"] = "Ring2";
    ItemEquipmentSlot["Ring3"] = "Ring3";
    ItemEquipmentSlot["Ring4"] = "Ring4";
    ItemEquipmentSlot["Pendant"] = "Pendant";
    ItemEquipmentSlot["Pendant2"] = "Pendant2";
    ItemEquipmentSlot["Weapon"] = "Weapon";
    ItemEquipmentSlot["Belt"] = "Belt";
    ItemEquipmentSlot["Hat"] = "Hat";
    ItemEquipmentSlot["FaceAccessory"] = "FaceAccessory";
    ItemEquipmentSlot["EyeAccessory"] = "EyeAccessory";
    ItemEquipmentSlot["Top"] = "Top";
    ItemEquipmentSlot["Bottom"] = "Bottom";
    ItemEquipmentSlot["Shoes"] = "Shoes";
    ItemEquipmentSlot["Earring"] = "Earring";
    ItemEquipmentSlot["ShoulderDecoration"] = "ShoulderDecoration";
    ItemEquipmentSlot["Glove"] = "Glove";
    ItemEquipmentSlot["Emblem"] = "Emblem";
    ItemEquipmentSlot["Badge"] = "Badge";
    ItemEquipmentSlot["Medal"] = "Medal";
    ItemEquipmentSlot["SecondaryWeapons"] = "SecondaryWeapons";
    ItemEquipmentSlot["Cape"] = "Cape";
    ItemEquipmentSlot["MechanicalHeart"] = "MechanicalHeart";
    ItemEquipmentSlot["PocketItem"] = "PocketItem";
})(ItemEquipmentSlot || (ItemEquipmentSlot = {}));
export var ItemEquipOption;
(function (ItemEquipOption) {
    ItemEquipOption["Str"] = "str";
    ItemEquipOption["Dex"] = "dex";
    ItemEquipOption["Int"] = "int";
    ItemEquipOption["Luk"] = "luk";
    ItemEquipOption["MaxHp"] = "max_hp";
    ItemEquipOption["MaxMp"] = "max_mp";
    ItemEquipOption["MaxHpRate"] = "max_hp_rate";
    ItemEquipOption["MaxMpRate"] = "max_mp_rate";
    ItemEquipOption["AttackPower"] = "attack_power";
    ItemEquipOption["MagicPower"] = "magic_power";
    ItemEquipOption["Armor"] = "armor";
    ItemEquipOption["BossDamage"] = "boss_damage";
    ItemEquipOption["IgnoreMonsterArmor"] = "ignore_monster_armor";
    ItemEquipOption["Damage"] = "damage";
    ItemEquipOption["Speed"] = "speed";
    ItemEquipOption["Jump"] = "jump";
    ItemEquipOption["AllStat"] = "all_stat";
    ItemEquipOption["EquipmentLevelDecrease"] = "equipment_level_decrease";
})(ItemEquipOption || (ItemEquipOption = {}));
export var SymbolEquipOption;
(function (SymbolEquipOption) {
    SymbolEquipOption["SymbolLevel"] = "symbol_level";
    SymbolEquipOption["Growth"] = "growth";
    SymbolEquipOption["Str"] = "symbol_str";
    SymbolEquipOption["Dex"] = "symbol_dex";
    SymbolEquipOption["Int"] = "symbol_int";
    SymbolEquipOption["Luk"] = "symbol_luk";
    SymbolEquipOption["Hp"] = "symbol_hp";
    SymbolEquipOption["ExpRate"] = "symbol_exp_rate";
    SymbolEquipOption["MesoRate"] = "symbol_meso_rate";
    SymbolEquipOption["DropRate"] = "symbol_drop_rate";
})(SymbolEquipOption || (SymbolEquipOption = {}));
