// @ruvice/my-maple-models/dist/models/maple/Equipment/equipmentTypes.js
export var ItemEquipmentSlot;
(function (ItemEquipmentSlot) {
    ItemEquipmentSlot["Ring1"] = "Ring1";
    ItemEquipmentSlot["Ring2"] = "Ring2";
    ItemEquipmentSlot["Ring3"] = "Ring3";
    ItemEquipmentSlot["Ring4"] = "Ring4";
    ItemEquipmentSlot["Pendant"] = "Pendant";
    ItemEquipmentSlot["Pendant2"] = "Pendant2";
    ItemEquipmentSlot["Weapon"] = "Weapon";
    ItemEquipmentSlot["Belt"] = "Belt";
    ItemEquipmentSlot["Hat"] = "Hat";
    ItemEquipmentSlot["FaceAccessory"] = "FaceAccessory";
    ItemEquipmentSlot["EyeAccessory"] = "EyeAccessory";
    ItemEquipmentSlot["Top"] = "Top";
    ItemEquipmentSlot["Bottom"] = "Bottom";
    ItemEquipmentSlot["Shoes"] = "Shoes";
    ItemEquipmentSlot["Earring"] = "Earring";
    ItemEquipmentSlot["ShoulderDecoration"] = "ShoulderDecoration";
    ItemEquipmentSlot["Glove"] = "Glove";
    ItemEquipmentSlot["Emblem"] = "Emblem";
    ItemEquipmentSlot["Badge"] = "Badge";
    ItemEquipmentSlot["Medal"] = "Medal";
    ItemEquipmentSlot["SecondaryWeapons"] = "SecondaryWeapons";
    ItemEquipmentSlot["Cape"] = "Cape";
    ItemEquipmentSlot["MechanicalHeart"] = "MechanicalHeart";
})(ItemEquipmentSlot || (ItemEquipmentSlot = {}));
